generators/javascript.js

```diff
@@ -169,11 +169,7 @@ Blockly.JavaScript.scrubNakedValue = function(line) {
  * @private
  */
 Blockly.JavaScript.quote_ = function(string) {
-  // TODO: This is a quick hack.  Replace with goog.string.quote
-  string = string.replace(/\\/g, '\\\\')
-                 .replace(/\n/g, '\\\n')
-                 .replace(/'/g, '\\\'');
-  return '\'' + string + '\'';
+  return goog.string.quote(string);
```

Comment:
The Google style guide (that we try to follow) recommends using single-quoted string literals in JavaScript:
https://google.github.io/styleguide/javascriptguide.xml?showone=Strings#Strings
goog.string.quote uses double-quotes.  Same issue applies to Dart.